src/application_input.hpp

```diff
@@ -52,7 +52,7 @@ struct ApplicationInput {
   // MeshBlock functions
   std::function<std::unique_ptr<MeshBlockApplicationData>(MeshBlock *, ParameterInput *)>
       InitApplicationMeshBlockData = nullptr;
-  std::function<void(ParameterInput *)> InitUserMeshBlockData = nullptr;
+  std::function<void(MeshBlock *, ParameterInput *)> InitUserMeshBlockData = nullptr;
```

Comment:
I think passing a pointer to the `MeshBlock` is the right thing to do here. But because we also have a `MeshBlockData` class, the name of the function might be a bit confusing (since it's user data for the `MeshBlock` and not the `MeshBlockData`). What about renaming the function to `InitMeshBlockUserData`?

Reply:
I like this idea. I'll make the change.